rfcs/0014-wal_on_disk_format.md

```diff
@@ -0,0 +1,293 @@
+This RFC only specifies the persistence format and some desirable properties of the in-memory data structure. 
+No exact implementation of the in-memory data structure is provided, so that the in-memory structure can be 
+modified without the need of new RFC.
+
+<!-- Briefly explain what this RFC changes and why. Prefer 3–6 sentences. -->
+
+<!-- TOC --><a name="motivation"></a>
+## Motivation
+
+Currently, the WAL is buffered in memory in a `KVTable`. 
+That is the same data structure that is used for the memtable. 
+The `KVTable` is a skip map -- an ordered map based on a skip list -- that keeps all incoming key-value pairs ordered by key. 
+While this is beneficial for the memtable for reading and flushing, it adds unnecessary overhead for the WAL.
+The WAL merely needs to maintain records in the order they are ingested to ensure durability and recovery.
+
+In object storage, the WAL is stored as a sorted string table (SST). 
+Encoding the SST from a `KVTable` also implies overhead that is unnecessary for the WAL. 
+For example, filters and indices are not needed for a WAL object where the records can be flushed to object 
+store in the same order they were ingested.
+
+For these reasons, the WAL can be implemented with a first-in first-out (FIFO) data structure in memory and with a persistence 
+format that sequentially stores and loads records without any specific indices or filters.   
+
+The overhead of using the skip map and the SST format for the WAL can be seen in the two flamegraphs included in 
+GitHub issue [#1085](https://github.com/slatedb/slatedb/issues/1085). The overhead hinders higher ingest throughput 
+into slateDB.
+
+<!-- What problem are we solving? What user or system pain exists today? Include concrete examples and why “do nothing” is insufficient. -->
+
+<!-- TOC --><a name="goals"></a>
+## Goals
+
+- Specify a queue-like format for WAL objects that can be flushed with minimal overhead.
+
+<!-- TOC --><a name="non-goals"></a>
+## Non-Goals
+
+- Specify the implementation of the in-memory data structure. 
+Merely, some desirable properties will be mentioned.  
+
+<!-- TOC --><a name="design"></a>
+## Design
+
+The proposed new persistence format for the WAL objects starts with a list of variable-length records.
+The list of records is followed by a list of record sizes represented by 4-bytes unsigned integers in little endian.
+The record sizes have the same order as the records. That is, the first record size is the size of the first record
+in the object, the second record size is the size of the second record in the object, and so on.
+After, the record sizes the format contains the number of records in the WAL object as a 4 bytes unsigned integer in little endian.
+Finally, the last 2 bytes contain the version of the format as an unsigned integer in little endian.
+
+
+```
++----------------------------------------------------------------+
+| record 0 (variable length)                                     |
```

Comment:
what is the structure of the record? just `Bytes`? I guess that would be tied to the in-memory implementation if we don't specify it. I assume things like the timestamp, the seqnum, etc... are all important metadata that we'll need.